modules/dnn/misc/python/test/test_dnn.py

```diff
@@ -135,6 +164,52 @@ def test_blobFromImage(self):
         target = target.transpose(2, 0, 1).reshape(1, 3, height, width)  # to NCHW
         normAssert(self, blob, target)
 
+    def test_model(self):
+        img_path = self.find_dnn_file("dnn/dog416.png")
+        weights = self.find_dnn_file("dnn/resnet50_rfcn_final.caffemodel")
+        config = self.find_dnn_file("dnn/rfcn_pascal_voc_resnet50.prototxt")
+        frame = cv.imread(img_path)
+        model = cv.dnn.DetectionModel_create(weights, config)
+        size = (800, 600)
+        mean = (102.9801, 115.9465, 122.7717)
+        model.setParams(width=size[0], height=size[1], mean=mean)
+
+        iouDiff = 0.011
+        confThreshold = 0.8
+        nmsThreshold = 0
+        scoreDiff = 4e-3
+
+        classIds, confidences, boxes = model.detect(frame, confThreshold, nmsThreshold)
+
+        refClassIds = (7, 12)
+        refConfidences = (0.991359, 0.94786)
+        refBoxes = ((491.822, 81.1668, 211.751, 98.0672),
+                    (132.093, 223.903, 206.984, 343.257))
+
+        normAssertDetections(self, refClassIds, refConfidences, refBoxes,
+                             classIds, confidences, boxes,
+                             confThreshold, scoreDiff, iouDiff)
+
+        frame = cv.resize(frame, size)
+        for box, conf, classId in zip(boxes, confidences, classIds):
+            left, top, width, height = box
+            left = int(left)
+            assert(left > 0)
+            top = int(top)
+            assert(top > 0)
+            width = int(width)
+            height = int(height)
+            assert(left + width < frame.shape[1])
+            assert(top + height < frame.shape[0])
+            cv.rectangle(frame, (left, top), (left + width, top + height), (0, 255, 0))
+
+            label = '%s: %.2f' % (classId, conf)
+
+            labelSize, baseLine = cv.getTextSize(label, cv.FONT_HERSHEY_SIMPLEX, 0.5, 1)
+            top = max(top, labelSize[1])
+            cv.rectangle(frame, (left, top - labelSize[1]), (left + labelSize[0], top + baseLine), (255, 255, 255), cv.FILLED)
+            cv.putText(frame, label, (left, top), cv.FONT_HERSHEY_SIMPLEX, 0.5, (0, 0, 0))
```

Comment:
Just the first `cv.rectangle` is enough. I just wanted to point that we need to cast coords to ints.